tiles/src/main/java/com/protomaps/basemap/layers/Boundaries.java

```diff
@@ -21,29 +21,213 @@ public String name() {
+          kind = "lease_limit";
+          admin_level = 3;
+        }
+        case "Line of control (please verify)" -> {
+          kind = "country";
+          kind_detail = "line_of_control";
+          disputed = true;
+        }
+        case "Overlay limit" -> {
+          kind = "overlay_limit";
+          admin_level = 3;
+        }
+        case "Unrecognized" -> kind = "unrecognized_country";
+        case "Map unit boundary" -> {
+          kind = "map_unit";
+          admin_level = 3;
+        }
+        case "Breakaway" -> {
+          kind = "unrecognized_country";
+          kind_detail = "disputed_breakaway";
+          admin_level = 3;
+        }
+        case "Claim boundary" -> {
+          kind = "unrecognized_country";
+          kind_detail = "disputed_claim";
+          admin_level = 3;
+        }
+        case "Elusive frontier" -> {
+          kind = "unrecognized_country";
+          kind_detail = "disputed_elusive";
+          admin_level = 3;
+        }
+        case "Reference line" -> {
+          kind = "unrecognized_country";
+          kind_detail = "disputed_reference_line";
+          admin_level = 3;
+        }
+        case "Admin-1 region boundary" -> {
+          kind = "macroregion";
+          admin_level = 3;
+        }
+        case "Admin-1 boundary" -> {
+          kind = "region";
+          admin_level = 4;
+        }
+        case "Admin-1 statistical boundary" -> {
+          kind = "region";
+          admin_level = 4;
+        }
+        case "Admin-1 statistical meta bounds" -> {
+          kind = "region";
+          admin_level = 4;
+        }
+        case "1st Order Admin Lines" -> {
+          kind = "region";
+          admin_level = 4;
+        }
+        case "Unrecognized Admin-1 region boundary" -> {
+          kind = "unrecognized_macroregion";
+          admin_level = 4;
+        }
+        case "Unrecognized Admin-1 boundary" -> {
+          kind = "unrecognized_region";
+          admin_level = 4;
+        }
+        case "Unrecognized Admin-1 statistical boundary" -> {
+          kind = "unrecognized_region";
+          admin_level = 4;
+        }
+        case "Unrecognized Admin-1 statistical meta bounds" -> {
+          kind = "unrecognized_region";
+          admin_level = 4;
+        }
+        default -> kind = "";
+      }
+    }
+
+    if (sf.canBeLine() && sf.hasTag("min_zoom") && (kind.equals("") == false && kind.equals("tz_boundary") == false)) {
+      features.line(this.name())
+        // Don't label lines to reduce file size (and they aren't shown in styles anyhow)
+        //.setAttr("name", sf.getString("name"))
+        .setAttr("pmap:min_zoom", sf.getLong("min_zoom"))
+        .setAttr("pmap:min_admin_level", admin_level)
+        .setZoomRange(
+          sf.getString("min_zoom") == null ? theme_min_zoom : (int) Double.parseDouble(sf.getString("min_zoom")),
+          theme_max_zoom)
+        .setAttr("pmap:ne_id", sf.getString("ne_id"))
```

Comment:
I would prefer to embed ne or OSM IDs in the 64-bit MVT id - are we ever going to use this for visualization or joining?

Reply:
For boundaries and `pmap:ne_id`, yes the intent (unlike with other layers) is to provide a fail safe during styling. This shouldn't be standard practice. Are the MVT IDs addressable in a MapLibre style?

Reply:
I see what you mean about `ne_id` being useful for failsafes, though it seems like referring to specific one-off features by name would be more natural. MVT IDs can addressed in the style, but in practice because I'm doing some bit twiddling the number is not going to be useful. I could be convinced to abandon MVT IDs if the storage doesn't inflate by storing all of source, ID, etc separately.

But it's important for us to have the OSM ID if possible to power debugging "where did this feature come from"

Reply:
Note that if we switch to daylight distribution and add buildings in the IDs all get renumbered and are worthless. (I too find them great for debug.)